### PR TITLE
feat(tql): add initial implementation for cte support

### PR DESCRIPTION
**Pull request by etolbakov**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

Support CTE queries for TQL

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)
https://github.com/GreptimeTeam/greptimedb/issues/1915

### CODE REVIEW

File: src/sql/src/parser.rs [hunk 1]
```diff
@@ -101,7 +101,20 @@ impl<'a> ParserContext<'a> {
...
                     Keyword::INSERT => self.parse_insert(),
 
-                    Keyword::SELECT | Keyword::WITH | Keyword::VALUES => self.parse_query(),
+                    Keyword::SELECT | Keyword::VALUES => self.parse_query(),
+
+                    Keyword::WITH => {
+                        let is_tql = self.parser.peek_nth_token(4);
+                        if is_tql.token.to_string().to_uppercase() == tql_parser::TQL {
+                            let parse_tql = self.parse_tql();
+                            dbg!(parse_tql.as_ref().clone());
```
Comment (waynexia) [inline src/sql/src/parser.rs:110]: we usually use `common_telemetry::debug!` to print debug log. And other temporary outputs need to be removed

File: src/sql/src/parsers/tql_parser.rs [hunk 2]
```diff
@@ -127,6 +155,38 @@ impl<'a> ParserContext<'a> {
...
+            match parser.peek_token().token {
+                Token::Word(w) if w.value.to_lowercase() == keyword_str => {
+                    break;
+                }
+                _ => parser.next_token(),
+            };
+        }
+    }
+
+    //TODO better name
```
Comment (waynexia) [inline src/sql/src/parsers/tql_parser.rs:170]: nit:
```suggestion
    // TODO better name
```

File: src/sql/src/parser.rs (hunk 1 above)
Comment (etolbakov) [reply src/sql/src/parser.rs:110]: yeah, make sense,👍  I would remove that debugging output anyway

**Comment by killme2008** [comment]
@etolbakov  Hi, still working on this PR?

**Comment by etolbakov** [comment]
hello, sorry I've stopped working on that

**Comment by killme2008** [comment]
> hello, sorry I've stopped working on that

All right. I closed this PR. Feel free to open it again when you want. Thanks.

**Pull request closed without merging**